UnityProject/Assets/Scripts/Chemistry/ReagentContainer.cs

```diff
@@ -376,19 +379,25 @@ private void SpillAll(Vector3Int worldPos, bool thrown = false)
 			foreach (var mob in mobs)
 			{
 				var mobGameObject = mob.gameObject;
-				Chat.AddCombatMsgToChat(mobGameObject, mobGameObject.name+" has been splashed with something!",
-					mobGameObject.name+" has been splashed with something!");
+				Chat.AddCombatMsgToChat(mobGameObject, mobGameObject.name + " has been splashed with something!",
+					mobGameObject.name + " has been splashed with something!");
 			}
 		}
 		else
 		{
-			Chat.AddLocalMsgToChat($"{gameObject.ExpensiveName()}'s contents spill all over the floor!",(Vector3)worldPos, gameObject);
+			Chat.AddLocalMsgToChat($"{gameObject.ExpensiveName()}'s contents spill all over the floor!", (Vector3)worldPos, gameObject);
 		}
 
 		var spilledReagents = TakeReagents(AmountOfReagents(Contents));
 		MatrixManager.ReagentReact(spilledReagents, worldPos);
 
-	//todo: half decent regs spread
+		var welder = gameObject.GetComponent<Welder>();
```

Comment:
this seems backwards. I’d rather introduce the reagent level UnityEvent in ReagentContainer and subscribe to it from within the welder script

Reply:
Will make your suggested change, i agree that would be a better idea.